go-controller/pkg/ovn/ovn.go

```diff
@@ -766,19 +774,37 @@ func (oc *Controller) WatchNetworkPolicy() {
 // WatchEgressFirewall starts the watching of egressfirewall resource and calls
 // back the appropriate handler logic
 func (oc *Controller) WatchEgressFirewall() *factory.Handler {
+	oc.triggerRetryObjs(oc.retryEgressFirewalls.retryChan, oc.iterateRetryEgressFirewalls)
 	return oc.watchFactory.AddEgressFirewallHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			egressFirewall := obj.(*egressfirewall.EgressFirewall).DeepCopy()
-			addErrors := oc.addEgressFirewall(egressFirewall)
-			if addErrors != nil {
-				klog.Error(addErrors)
-				egressFirewall.Status.Status = egressFirewallAddError
-			} else {
-				egressFirewall.Status.Status = egressFirewallAppliedCorrectly
+			key := getEgressFirewallNamespacedName(egressFirewall)
+			oc.retryEgressFirewalls.initRetryObjWithAdd(egressFirewall, key)
+			oc.retryEgressFirewalls.skipRetryObj(key)
+			egressFirewall.Status.Status = egressFirewallAppliedCorrectly
+			if retryEntry := oc.retryEgressFirewalls.getObjRetryEntry(key); retryEntry != nil && retryEntry.oldObj != nil {
+				klog.Infof("Detected stale egress firewall during new one add with the same name: %s", key)
+				egressFirewall := retryEntry.oldObj.(*egressfirewall.EgressFirewall)
+				if err := oc.deleteEgressFirewall(egressFirewall); err != nil {
+					oc.retryEgressFirewalls.unSkipRetryObj(key)
+					klog.Errorf("Failed to delete stale egress firewall %s, during add: %v", key, err)
+					egressFirewall.Status.Status = egressFirewallAddError
+				} else {
+					oc.retryEgressFirewalls.removeDeleteFromRetryObj(key)
+				}
 			}
-
-			err := oc.updateEgressFirewallWithRetry(egressFirewall)
-			if err != nil {
+			if egressFirewall.Status.Status != egressFirewallAddError {
+				start := time.Now()
+				if err := oc.addEgressFirewall(egressFirewall); err != nil {
+					klog.Errorf("Failed to create egress firewall %s, error: %v", key, err)
+					egressFirewall.Status.Status = egressFirewallAddError
+					oc.retryEgressFirewalls.unSkipRetryObj(key)
+				} else {
+					oc.retryEgressFirewalls.deleteRetryObj(key, true)
+					klog.Infof("Created Egress firewall: %s took: %v", key, time.Since(start))
```

Comment:
Should we also log how long it takes to update and delete an egressFirewall? or do we only care about creation time? on that note would it be useful to know how long it took for the egressFirewall creation to fail and add the timestamp to the error? just thinking out loud feel free to ignore

Reply:
I will log both update and delete I think this is a good idea